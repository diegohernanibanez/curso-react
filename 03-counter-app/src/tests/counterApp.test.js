import React from 'react';
import CounterApp from '../CounterApp';
import {shallow} from 'enzyme';

describe('Prubas en el archivo counterApp', () => {
    
    test('deben de ser iguales los strings', () => {
        const msg = "Hola mundo";
    
        const msg2 = `Hola mundo`;
    
        expect(msg).toBe(msg2);
    });

    test('debe mostrar CounterApp correctamente', () => {
        const wrapper = shallow(<CounterApp />);

        expect(wrapper).toMatchSnapshot(); 
    });

    test('debe mostrar el CounterApp con valor por defecto 100', () => {
        const valor = 100;

        const wrapper = shallow(<CounterApp value = {valor} />);

        const valorMostrado = wrapper.find('h2').text().trim();
        
        expect(valorMostrado).toBe(valor.toString());
    })
    
    
});


