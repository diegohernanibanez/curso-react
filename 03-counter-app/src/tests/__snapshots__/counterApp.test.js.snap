// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas en el archivo counterApp debe mostrar CounterApp correctamente 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
